rotkehlchen/serialization/schemas.py

```diff
@@ -102,24 +97,74 @@ def validate_schema(
             data: dict[str, Any],
             **_kwargs: Any,
     ) -> None:
-        if self.expected_asset_type is None and data['asset_type'] is None:
-            raise ValidationError(message='asset_type should be given', field_name='asset_type')
-        if self.expected_asset_type is not None and data['asset_type'] is not None and data['asset_type'] != self.expected_asset_type:  # noqa: E501
-            raise ValidationError(message=f'asset_type should be {self.expected_asset_type}', field_name='asset_type')  # noqa: E501
         if self.identifier_required is True and data['identifier'] is None:
             raise ValidationError(message='Asset identifier should be given', field_name='identifier')  # noqa: E501
         if self.coingecko_obj is not None:
             _validate_single_oracle_id(data, 'coingecko', self.coingecko_obj)
         if self.cryptocompare_obj is not None:
             _validate_single_oracle_id(data, 'cryptocompare', self.cryptocompare_obj)
 
+
+class FiatAssetSchema(AssetWithOraclesSchema):
+    @post_load
+    def transform_data(self, data: dict[str, Any], **_kwargs: Any) -> FiatAsset:
+        return FiatAsset.initialize(
+            # asset id needs to be unique but no combination of asset data is guaranteed to be
+            # unique. And especially with the ability to edit assets we need an external uuid.
+            # TODO: Asset uuid is generated only for crypto and fiat assets, but we should figure
+            # out a way to generate the identifier out of the asset fields, same as we do for
+            # evm tokens.
+            identifier=data['identifier'] if data['identifier'] is not None else str(uuid4()),
+            name=data['name'],
+            symbol=data['symbol'],
+            coingecko=data['coingecko'],
+            cryptocompare=data['cryptocompare'],
+        )
+
+
+class CryptoAssetFieldsSchema(Schema):
+    started = TimestampField(load_default=None)
+    swapped_for = AssetField(expected_type=CryptoAsset, load_default=None)
+    forked = AssetField(expected_type=CryptoAsset, load_default=None)
+
+
+class CryptoAssetSchema(AssetWithOraclesSchema, CryptoAssetFieldsSchema):
+    asset_type = AssetTypeField(load_default=None)
```

Comment:
Could you add docstrings to this functions? So it is easier to know why you have something that is fields and another that has only crypto asset. At first is hard to follow